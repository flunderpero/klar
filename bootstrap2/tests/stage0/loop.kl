fn while_loop():
    let i = 0
    while i < 3:
        i = i + 1
        print(f"regular: {i}")
    end
    i = 0
    while i < 3:
        i = i + 1
        print(f"with break: {i}")
        if i == 2 => break
    end
    i = 0
    while i < 3:
        i = i + 1
        if i == 1 => continue
        print(f"with continue: {i}")
    end
end

fn main():
    while_loop()
end
