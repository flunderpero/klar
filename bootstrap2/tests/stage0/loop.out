regular: 1
regular: 2
regular: 3
with break: 1
with break: 2
with continue: 2
with continue: 3
